From 03dc7e49433defbe47b0c900c86b0968eeda174c Mon Sep 17 00:00:00 2001
From: Damjan Georgievski <gdamjan@gmail.com>
Date: Tue, 10 Jan 2017 01:20:25 +0100
Subject: [PATCH] fix compilation with php 7.1

php 7.1 changed the signature of the log_message function in
sapi_module_struct. Let's assume it will stay like this for all versions >= 7.1

fixes #1427
---
 plugins/php/php_plugin.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/plugins/php/php_plugin.c b/plugins/php/php_plugin.c
index 7440d55..78d980f 100644
--- a/plugins/php/php_plugin.c
+++ b/plugins/php/php_plugin.c
@@ -568,8 +568,11 @@ static int php_uwsgi_startup(sapi_module_struct *sapi_module)
 	}
 }
 
+#if ((PHP_MAJOR_VERSION >= 7) && (PHP_MINOR_VERSION >= 1))
+static void sapi_uwsgi_log_message(char *message, int syslog_type_int) {
+#else
 static void sapi_uwsgi_log_message(char *message TSRMLS_DC) {
-
+#endif
 	uwsgi_log("%s\n", message);
 }
 
